src/main/java/org/dcsa/core/exception/handler/GlobalExceptionHandler.java

```diff
@@ -24,42 +25,66 @@
 public class GlobalExceptionHandler {
 
   @ExceptionHandler(ConcreteRequestErrorMessageException.class)
-  public ResponseEntity<RequestFailureTO> handle(ServerHttpRequest serverHttpRequest, ConcreteRequestErrorMessageException ex) {
+  public ResponseEntity<RequestFailureTO> handle(
+      ServerHttpRequest serverHttpRequest, ConcreteRequestErrorMessageException ex) {
     ResponseStatus responseStatusAnnotation = ex.getClass().getAnnotation(ResponseStatus.class);
     HttpStatus httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
     ConcreteRequestErrorMessageTO errorEntity = ex.asConcreteRequestMessage();
     if (responseStatusAnnotation != null) {
       httpStatus = responseStatusAnnotation.value();
     }
-    RequestFailureTO failureTO = new RequestFailureTO(
+    RequestFailureTO failureTO =
+        new RequestFailureTO(
             serverHttpRequest.getMethodValue(),
             serverHttpRequest.getURI().toString(),
             List.of(errorEntity),
-            httpStatus
-    );
+            httpStatus);
     return new ResponseEntity<>(failureTO, httpStatus);
   }
 
   @ExceptionHandler(DCSAException.class)
-  public void handleDCSAExceptions(DCSAException dcsaEx) {
+  public ResponseEntity<RequestFailureTO> handleDCSAExceptions(
+      ServerHttpRequest serverHttpRequest, DCSAException dcsaEx) {
+
     log.debug(
         "{} ({}) - {}",
         this.getClass().getSimpleName(),
         dcsaEx.getClass().getSimpleName(),
         dcsaEx.getMessage());
     logExceptionTraceIfEnabled(dcsaEx);
-    throw dcsaEx;
+
+    ResponseStatus responseStatusAnnotation = dcsaEx.getClass().getAnnotation(ResponseStatus.class);
+
+    HttpStatus httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
+
+    if (responseStatusAnnotation != null) {
+      httpStatus = responseStatusAnnotation.value();
+    }
+
+    ConcreteRequestErrorMessageTO errorEntity =
+        new ConcreteRequestErrorMessageTO("internal_error", dcsaEx.getMessage());
```

Comment:
I think this should be `internalError` (based on the Design Principles, we use camelCase for these values).

Also, I find it strange that we hard-code it to `internalError` when the status can be a BAD_REQUEST (via the `responseStatusAnnotation`)

Reply:
Made the changes, also changed the hard-coded value to the reason phrase provided by the HTTP Status thrown.